examples/download_progress/src/main.rs

```diff
@@ -1,144 +1,277 @@
+                    .spacing(20)
+                    .push(
+                        Text::new("Download Progress")
+                            .width(Length::Fill)
+                            .size(40)
+                            .horizontal_alignment(HorizontalAlignment::Center)
+                            .color(Color::from_rgb8(16, 93, 208)),
+                    )
+                    .push(
+                        Text::new("Loading...")
+                            .width(Length::Fill)
+                            .size(28)
+                            .horizontal_alignment(HorizontalAlignment::Center)
+                            .color(Color::from_rgb(0.3, 0.3, 0.3)),
+                    ),
+            )
+            .width(Length::Fill)
+            .center_x()
+            .into(),
+            Example::Loaded(State {
+                tasks, scrollable, ..
+            }) => {
+                let list = tasks.iter_mut().fold(
+                    Column::new()
+                        .width(Length::Fill)
+                        .spacing(20)
+                        .align_items(Align::Center),
+                    |column, task| column.push(task.view()),
+                );
+                Container::new(
+                    Scrollable::new(scrollable).padding(40).spacing(40).align_items(Align::Center)
+                        .push(Text::new("Download multiple files asynchronously, click again to cancel the download task.")
+                            .horizontal_alignment(HorizontalAlignment::Center)
+                            .width(Length::Units(400)))
+                        .push(list),
                 )
-                .into(),
-            Example::Downloading { .. } => {
-                Text::new(format!("Downloading... {:.2}%", current_progress))
+                    .width(Length::Fill)
+                    .center_x()
                     .into()
             }
-            Example::Errored { button } => Column::new()
-                .spacing(10)
-                .align_items(Align::Center)
-                .push(Text::new("Something went wrong :("))
-                .push(
-                    Button::new(button, Text::new("Try again"))
-                        .on_press(Message::Download),
-                )
-                .into(),
-        };
-
-        let content = Column::new()
-            .spacing(10)
-            .padding(10)
-            .align_items(Align::Center)
-            .push(progress_bar)
-            .push(control);
-
-        Container::new(content)
-            .width(Length::Fill)
-            .height(Length::Fill)
-            .center_x()
-            .center_y()
-            .into()
+        }
     }
 }
+
+fn main() {
+    Example::run(Settings {
+        window: window::Settings {
+            size: (800, 600),
+            resizable: true,
+            decorations: true,
+        },
+        flags: (),
+        default_font: None,
+        antialiasing: false,
+    });
+}
```

Comment:
All the examples have `main` at the beginning, followed by the application state and trait implementation. We should also use the default `Settings`.